blocks/monitoring/aggregation.py

```diff
@@ -103,14 +104,20 @@ def __init__(self, numerator, denominator):
         self.denominator = denominator
 
     def get_aggregator(self):
+        initialized = shared_like(0.)
         numerator_acc = shared_like(self.numerator)
         denominator_acc = shared_like(self.denominator)
-        initialization_updates = [(numerator_acc, 0.0),
-                                  (denominator_acc, 0.0)]
+        conditional_update = ifelse(initialized,
+                                    self.numerator + numerator_acc,
+                                    self.numerator)
+        initialization_updates = [(numerator_acc, tensor.zeros_like(numerator_acc)),
+                                  (denominator_acc, 0.0),
```

Comment:
We could allow tensor denominator by using `tensor.zeros_like(denominator_acc)`, couldn't we? I can hardly ever imagine who might need it, but it is nice to have complete generality when it comes almost for free, I think.